provider/openstack: upgrade asg operations to use lbaasv2

drmaas opened the pull request.

Commit:
provider/openstack: upgrade asg operations to use lbaasv2

tomaslin committed:
set inservice flag on new cluster creation requests

varikin commented on clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/deploy/description/servergroup/ServerGroupParameters.groovy:
```diff
@@ -95,6 +93,24 @@ class ServerGroupParameters {
   /**
+   * Stack parameters of type 'comma_delimited_list' come back as a unicode json string. We need to split that up.
+   *
+   * I need a shower.
```
May not want to leave this comment in

tomaslin committed:
Merge pull request #892 from tomaslin/titus-inservice-create

set inservice flag on new cluster creation requests

derekolk commented on clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/deploy/description/servergroup/MemberData.groovy:
```diff
@@ -0,0 +1,26 @@
+package com.netflix.spinnaker.clouddriver.openstack.deploy.description.servergroup
+
+import groovy.transform.Canonical
+
```
Can you please provide context?  Why and what is it used for?

derekolk commented on clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/deploy/ops/servergroup/DeployOpenstackAtomicOperation.groovy:
```diff
@@ -112,28 +107,37 @@ class DeployOpenstackAtomicOperation implements AtomicOperation<DeploymentResult
-      task.updateStatus BASE_PHASE, "Getting internal port used for load balancer $pool.name"
-      int port = provider.getInternalLoadBalancerPort(pool)
-      task.updateStatus BASE_PHASE, "Found internal port $port used for load balancer $pool.name"
+      //look up all load balancer listeners -> pool ids and internal ports
```
Can this be functionality be pulled out into a common method shared across upsertServerGroup/resizeServerGroup/upsertLoadBalancer?

derekolk commented on clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/deploy/ops/servergroup/ServerGroupConstants.groovy:
```diff
@@ -0,0 +1,39 @@
+
+package com.netflix.spinnaker.clouddriver.openstack.deploy.ops.servergroup
+
+/**
```
Remove empty comment or provide a definition.

derekolk commented on clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/deploy/ops/servergroup/ServerGroupConstants.groovy:
```diff
@@ -0,0 +1,39 @@
+  //and is contructed on the fly
+  final static String MEMBERTEMPLATE_FILE = 'asg_member.yaml'
+
+  final static String SUBTEMPLATE_OUTPUT = 'asg_resource'
```
If the output and files are dependent on each other, then it makes sense to use the OUTPUT with the FILE definition.

varikin committed:
Added credentials alias on openstack operations. (#893)

Deck wants to send the account name as 'credentials' which has to be
overriden for every openstack operation done in deck. Switching to the
standard pattern.

jtk54 committed:
provider/google: Support for deleting Http(s) load balancers. (#894)

Also included safe delete for network load balancers and
optionally not deleting GCP health checks.

gregturn committed:
provider/cf: Add 'disk limit' as an option for deployment (#897)

gregturn committed:
provider/cf: Find buildInfo information with safe navigation (#900)

Commit:
provider/google: Use pagination when retrieving instances. (#901)

Commit:
provider/openstack: unignore critical attributes in json responses (#902)

danielpeach committed:
provider/google updated gceVCpuMaxList (#891)

varikin committed:
Ensuring the scaling config serialized to numbers. (#890)

The generic map of options had the maxSize, minSize, etc as strings. Now
converting to stringified ints to real ints.

Added subnetId to server group detail for Openstack.

Commit:
provider/openstack: add desired field to server group capacity object (#904)
